sktime/transformers/series_as_features/dictionary_based/_sax.py

```diff
@@ -185,10 +190,19 @@ def push(self, letter):
 
     def shorten(self, amount):
         # shorten a word by set amount of letters
-        self.word = self.right_shift(self.word, amount * 2)
+        _BitWord.shorten_word(self.word, amount)
```

Comment:
why not use self?
```suggestion
        self.shorten_word(self.word, amount)
```
here and below in `shorten_word()` too, also not sure why making it a static method but then passing in `self.word` but I may not understand the full context here, perhaps leave a short comment to clear this up :)

Reply:
Very good spot on :) 

My main reason was to get rid of the _BitWord representation within BOSS and WEASEL. Thus these calls to _BitWord are static from outside the class now and I do only store lists of "int"-representation of the words. I though this safes some memory compared to storing lists of BitWords, which was the old representation.

To maintain backward-compatibility, I didnt want to remove the old methods though - maybe make them deprecated? What do you think?

Reply:
I'm not too worried about backward compatibility at this point, a lot of our API is still somewhat experimental :)

Reply:
I see :)

the BitWord instances were never really used other than as an intermediate step. I have removed that intermediate step. Therefor, I have removed the member variables of the BitWord and made it a static library + refactored SAX and SFA.